programs/ssl/ssl_server2.c

```diff
@@ -1838,7 +1838,10 @@ int main( int argc, char *argv[] )
 
 #if defined(MBEDTLS_MEMORY_BUFFER_ALLOC_C)
     mbedtls_memory_buffer_alloc_init( alloc_buf, sizeof(alloc_buf) );
-#endif
+#if defined(MBEDTLS_MEMORY_DEBUG)
+    size_t current_heap_memory, current_heap_blocks;
```

Comment:
Do we not want to measure the peak heap usage as well, to let us know if the RAM usage decreased after the handshake completed?

Reply:
Yes, I think both of these measurements could give us more information about progress of optimization. I will add it. Should I add peak memory checking to test also?

Reply:
I don't think we need the peak usage right now, actually. All the tests we want to do right now for acceptance are with post-handshake current memory usage, not peak.

1. Without MFL, what is the server’s post-handshake memory usage?
2. With ssl_client2 max_frag_len=512, what is the server’s  post-handshake memory usage?
3. With ssl_client2 max_frag_len=1024, what is the server’s  post-handshake memory usage?
4. With ssl_client2 max_frag_len=2048, what is the server’s post-handshake memory usage?
5. With ssl_client2 max_frag_len=4096, what is the server’s post-handshake memory usage?

Reply:
I added peak to print. Now its looks "Heap memory usage after handshake: 56024 bytes. In peak was 79436"